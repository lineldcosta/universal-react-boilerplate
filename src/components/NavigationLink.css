.navigationLink {
  color: palevioletred;
  text-decoration: none;
  padding: 0.5em;
  border-radius: 3px;
}

.active {
  color: white;
  background: palevioletred;
}

.listItem {
  list-style: none;
  display: inline-block;
  margin-right: 1em;
}

.listItem:last-child {
  margin: 0;
}
