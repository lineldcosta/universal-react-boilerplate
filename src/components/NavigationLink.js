/* @flow */
import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import styles from './NavigationLink.css'

import type { Element } from 'react'

type Props = {
  to: string,
  exact?: boolean,
  children?: Element<*>
}

const NavigationLink = ({ to, exact, children }: Props) => (
  <li className={styles.listItem}>
    <NavLink
      to={to}
      exact={exact}
      className={styles.navigationLink}
      activeClassName={styles.active}>{ children }</NavLink>
  </li>
)

NavigationLink.propTypes = {
  to: PropTypes.string.isRequired,
  exact: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string
  ])
}

export default NavigationLink
