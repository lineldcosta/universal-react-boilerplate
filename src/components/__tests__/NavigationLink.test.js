/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { StaticRouter } from 'react-router-dom'
import NavigationLink from '../NavigationLink'

const setup = (propOverrides, render = shallow) => {
  const props = Object.assign({
    to: '/test',
    location: '/'
  }, propOverrides)

  const router = render(
    <StaticRouter location={props.location} context={{}}>
      <NavigationLink {...props} />
    </StaticRouter>
  )
  const wrapper = router.find(NavigationLink)

  return {
    props,
    wrapper,
    router,
    link: router.find('a')
  }
}

it('renders correctly', () => {
  const { wrapper } = setup()
  expect(wrapper).toMatchSnapshot()
})

it('includes a link to the specified URL', () => {
  const to = '/a/test/url'
  const { link } = setup({ to }, mount)

  expect(link.prop('href')).toBe(to)
})
