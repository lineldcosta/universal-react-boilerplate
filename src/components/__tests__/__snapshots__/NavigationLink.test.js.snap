// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <NavigationLink
    location="/"
    to="/test"
/>,
  "nodes": Array [
    <NavigationLink
      location="/"
      to="/test"
/>,
  ],
  "options": Object {},
  "renderer": null,
  "root": ShallowWrapper {
    "complexSelector": ComplexSelector {
      "buildPredicate": [Function],
      "childrenOfNode": [Function],
      "findWhereUnwrapped": [Function],
    },
    "length": 1,
    "node": <Router
      history={
            Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "y2ktlo",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "y2ktlo",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
      }
>
      <NavigationLink
            location="/"
            to="/test"
      />
</Router>,
    "nodes": Array [
      <Router
        history={
                Object {
                        "action": "POP",
                        "block": [Function],
                        "canGo": [Function],
                        "createHref": [Function],
                        "entries": Array [
                          Object {
                            "hash": "",
                            "key": "y2ktlo",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                        ],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "index": 0,
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "key": "y2ktlo",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
        }
>
        <NavigationLink
                location="/"
                to="/test"
        />
</Router>,
    ],
    "options": Object {},
    "renderer": ReactShallowRenderer {
      "_instance": ShallowComponentWrapper {
        "_calledComponentWillUnmount": false,
        "_compositeType": 0,
        "_context": Object {},
        "_currentElement": <MemoryRouter
          initialEntries={
                    Array [
                              "/",
                            ]
          }
>
          <NavigationLink
                    location="/"
                    to="/test"
          />
</MemoryRouter>,
        "_debugID": 1,
        "_hostContainerInfo": null,
        "_hostParent": null,
        "_instance": MemoryRouter {
          "_reactInternalInstance": [Circular],
          "context": Object {},
          "history": Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "y2ktlo",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "y2ktlo",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          },
          "props": Object {
            "children": <NavigationLink
              location="/"
              to="/test"
/>,
            "initialEntries": Array [
              "/",
            ],
          },
          "refs": Object {},
          "state": null,
          "updater": Object {
            "enqueueCallback": [Function],
            "enqueueCallbackInternal": [Function],
            "enqueueElementInternal": [Function],
            "enqueueForceUpdate": [Function],
            "enqueueReplaceState": [Function],
            "enqueueSetState": [Function],
            "isMounted": [Function],
            "validateCallback": [Function],
          },
        },
        "_mountOrder": 1,
        "_pendingCallbacks": null,
        "_pendingElement": null,
        "_pendingForceUpdate": false,
        "_pendingReplaceState": false,
        "_pendingStateQueue": null,
        "_renderedComponent": NoopInternalComponent {
          "_currentElement": <Router
            history={
                        Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "canGo": [Function],
                                    "createHref": [Function],
                                    "entries": Array [
                                      Object {
                                        "hash": "",
                                        "key": "y2ktlo",
                                        "pathname": "/",
                                        "search": "",
                                        "state": undefined,
                                      },
                                    ],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "index": 0,
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "key": "y2ktlo",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
            }
>
            <NavigationLink
                        location="/"
                        to="/test"
            />
</Router>,
          "_debugID": 2,
          "_renderedOutput": <Router
            history={
                        Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "canGo": [Function],
                                    "createHref": [Function],
                                    "entries": Array [
                                      Object {
                                        "hash": "",
                                        "key": "y2ktlo",
                                        "pathname": "/",
                                        "search": "",
                                        "state": undefined,
                                      },
                                    ],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "index": 0,
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "key": "y2ktlo",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
            }
>
            <NavigationLink
                        location="/"
                        to="/test"
            />
</Router>,
        },
        "_renderedNodeType": 1,
        "_rootNodeID": 0,
        "_topLevelWrapper": null,
        "_updateBatchNumber": null,
        "_warnedAboutRefsInRender": false,
      },
      "getRenderOutput": [Function],
      "render": [Function],
    },
    "root": [Circular],
    "unrendered": <MemoryRouter
      initialEntries={
            Array [
                  "/",
                ]
      }
>
      <NavigationLink
            location="/"
            to="/test"
      />
</MemoryRouter>,
  },
  "unrendered": null,
}
`;
